waiter/src/waiter/request_log.clj

```diff
@@ -0,0 +1,106 @@
+;;
+;;       Copyright (c) 2018 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.request-log
+  (:require [clj-time.core :as t]
+            [clojure.core.async :as async]
+            [clojure.data.json :as json]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [waiter.async-utils :as au]
+            [waiter.ring-utils :as ru]
+            [waiter.utils :as utils]))
+
+(defn log
+  "Log log-data as JSON"
+  [log-data]
+  (log/info (json/write-str log-data :escape-slash false)))
+
+(defn request->context
+  "Convert a request into a context suitable for logging."
+  [{:keys [headers query-string remote-addr request-id request-method request-time uri] :as request}]
+  (let [{:strs [host x-cid x-forwarded-for]} headers]
+    (cond-> {:cid x-cid
+             :host host
+             :path uri
+             :remote-addr (or x-forwarded-for remote-addr)
+             :request-id request-id
+             :scheme (let [scheme (utils/request->scheme request)]
+                       (if (keyword? scheme)
+                         (name scheme)
+                         scheme))}
+      request-method (assoc :method (-> request-method name str/upper-case))
+      query-string (assoc :query-string query-string)
+      request-time (assoc :request-time (utils/date-to-str request-time)))))
+
+(defn response->context
+  "Convert a response into a context suitable for logging."
+  [{:keys [authorization/principal backend-response-latency-ms descriptor instance
+           get-instance-latency-ms response-body-size status total-latency-ms] :as response}]
+  (let [{:keys [service-id service-description]} descriptor]
+    (cond-> {:status (or status 200)}
+      backend-response-latency-ms (assoc :backend-response-latency-ms backend-response-latency-ms)
+      descriptor (assoc :metric-group (get service-description "metric-group")
+                        :service-id service-id
+                        :service-name (get service-description "name")
+                        :service-version (get service-description "version"))
+      instance (assoc :instance-host (:host instance)
+                      :instance-id (:id instance)
+                      :instance-port (:port instance)
+                      :instance-proto (:protocol instance)
+                      :get-instance-latency-ms get-instance-latency-ms)
+      principal (assoc :principal principal)
+      response-body-size (assoc :response-body-size response-body-size)
+      total-latency-ms (assoc :total-latency-ms total-latency-ms))))
+
+(defn stream-countable-body!
```

Comment:
`stream-http-response` has `(au/timed-offer! resp-chan buffer streaming-timeout-ms)`. Should that logic be repeated here?

Reply:
No longer counting bytes.